errand_matcher/static/errand_matcher/complete-signup.js

```diff
@@ -1,10 +1,75 @@
 // complete-signup.js
 // script for 'complete_signup' endpoint
 
+// Keeps track of which page is being viewed.
+var pageIndex;
+
+// Google Maps autocomplete widget
+var autocomplete;
+
+// Bias the autocomplete object to the user's geographical location,
+// as supplied by the browser's 'navigator.geolocation' object.
+function geolocate() {
+  if (navigator.geolocation) {
+    navigator.geolocation.getCurrentPosition(function(position) {
+      var geolocation = {
+        lat: position.coords.latitude,
+        lng: position.coords.longitude
+      };
+      var circle = new google.maps.Circle(
+          {center: geolocation, radius: position.coords.accuracy});
+      autocomplete.setBounds(circle.getBounds());
+    });
+  }
+}
+
+// // Initialize the Google Maps API autocomplete widget for address entry.
+function initAutocomplete() {
+  // Create the autocomplete object, restricting the search predictions to
+  // geographical location types.
+
+	autocomplete = new google.maps.places.Autocomplete(
+		document.getElementById('address-input'), {types: ['geocode']});
+
+	// Avoid paying for data that you don't need by restricting the set of
+	// place fields that are returned to just the address components.
+	autocomplete.setFields(['address_component','geometry']);
+
+	// When the user selects an address from the drop-down, populate the
+	// address fields in the form.
+	autocomplete.addListener('place_changed', function(){
+
+		var place = autocomplete.getPlace();
+		var zip;
+
+		$.each(place ? place.address_components : [], function(){
+			if (this.types[0] == "postal_code") {
+				zip = this.long_name;
+			}
+		});
+
+
+		// SV Question: What should we do if address is provided with no ZIP code (but valid lat/lng)?
+		if (!place || !place.geometry || !place.address_components || !zip) {
```

Comment:
Currently only allows addresses/places that Google associates with a specific zip-code. For example, "Brookline, MA" doesn't work. Not sure if this is desired behavior.